### PR TITLE
[ACR] Add blob/manifest operations samples to readme

### PR DESCRIPTION
**Pull request by YalinLi0312**
For GA consistency check in issue https://github.com/Azure/azure-sdk-for-python/issues/27343

Updates:
- Add blob/manifest samples to README.md
- Also catch `ServiceResponseError` that occurred when upload blob tests failed as service not supporting retries on upload blob for now.
- Use default audience in samples
- The total running time in USGov and China clouds are usually longer than the maximum 120 mins. To resolve this issue,
  - merged some tests so that they can share the same resources (e.g. image, blob)
  - load less images at the beginning of running tests

### CODE REVIEW

**Review by kashifkhan — APPROVED** [review]

File: sdk/containerregistry/azure-containerregistry/README.md [hunk 1]
```diff
@@ -137,6 +143,105 @@ with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https
                 client.delete_manifest(repository, manifest.digest)
 ```
 
+### Upload images
+
+To upload a full image, we need to upload individual layers and configuration. After that we can upload a manifest which describes an image or artifact and assign it a tag.
+
```
Comment (xiangyan99) [inline sdk/containerregistry/azure-containerregistry/README.md:149]: Can you add the snippet annotation like : \?

Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/README.md:156]: Updated

File: sdk/containerregistry/azure-containerregistry/README.md [hunk 2]
```diff
@@ -69,74 +69,218 @@ For more information please see [Container Registry Concepts](https://docs.micro
...
 
 Delete images older than the first three in the repository.
 
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_delete_images.delete_manifests -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
+```python
+with ContainerRegistryClient(self.endpoint, self.credential) as client:
```
Comment (xiangyan99) [inline sdk/containerregistry/azure-containerregistry/README.md:144]: Please double check if we need to provide audience in the ctor?

File: sdk/containerregistry/azure-containerregistry/README.md [hunk 3]
```diff
@@ -69,74 +69,218 @@ For more information please see [Container Registry Concepts](https://docs.micro
...
-```python
-endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
+<!-- SNIPPET:sample_list_tags.list_tags_anonymous -->
 
-with ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="https://management.azure.com") as client:
-    manifest = client.get_manifest_properties("library/hello-world", "latest")
-    print(manifest.repository_name + ": ")
+```python
+endpoint = os.environ.get("CONTAINERREGISTRY_ANONREGISTRY_ENDPOINT")
+with ContainerRegistryClient(endpoint) as anon_client:
```
Comment (xiangyan99) [inline sdk/containerregistry/azure-containerregistry/README.md:108]: Do we need to provide credential and audience? (I want to confirm if we need to update samples)

File: sdk/containerregistry/azure-containerregistry/README.md (hunk 2 above)
Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/README.md:144]: No need. - We'll use the default audience when not providing, and we have changed the default value to one that works across clouds.

File: sdk/containerregistry/azure-containerregistry/README.md (hunk 3 above)
Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/README.md:108]: This is anonymous access sample, credential is expected to be `None`.

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**